sdk/core/Azure.Core.TestFramework/src/TestRecording.cs

```diff
@@ -19,44 +22,110 @@ public class TestRecording : IDisposable
         private const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
         // cspell: disable-next-line
         private const string charsLower = "abcdefghijklmnopqrstuvwxyz0123456789";
+        private const string Sanitized = "Sanitized";
         internal const string DateTimeOffsetNowVariableKey = "DateTimeOffsetNow";
 
-        public TestRecording(RecordedTestMode mode, string sessionFile, RecordedTestSanitizer sanitizer, RecordMatcher matcher)
+        public SortedDictionary<string, string> Variables => _useLegacyTransport ? Session.Variables : _variables;
+        private readonly SortedDictionary<string, string> _variables = new();
+
+        public TestRecording(RecordedTestMode mode, string sessionFile, RecordedTestSanitizer sanitizer, RecordMatcher matcher, int? proxyPortHttp = default, int? proxyPortHttps = default, bool useLegacyTransport = false)
         {
             Mode = mode;
             _sessionFile = sessionFile;
             _sanitizer = sanitizer;
             _matcher = matcher;
 
-            switch (Mode)
+            _useLegacyTransport = useLegacyTransport;
+            ProxyPortHttp = proxyPortHttp;
+            ProxyPortHttps = proxyPortHttps;
+            var options = new TestProxyClientOptions();
+            var handler = new HttpClientHandler
             {
-                case RecordedTestMode.Record:
-                    Session = new RecordSession();
-                    if (File.Exists(_sessionFile))
-                    {
+                ServerCertificateCustomValidationCallback = (_, certificate, _, _) => certificate.Issuer == "CN=localhost"
+            };
+            options.Transport = new HttpClientTransport(handler);
+            ProxyClient = new TestProxyRestClient(new ClientDiagnostics(options), HttpPipelineBuilder.Build(options), new Uri($"https://127.0.0.1:{proxyPortHttps}"));
```

Comment:
Let's move all test-proxy process management code into a separate class and put ProxyClient there as well. We don't need a new one for every test.

Reply:
Also, use http port for management and you won't need the cert override